src/main/groovy/de/dkfz/roddy/tools/DateTimeHelper.groovy

```diff
@@ -0,0 +1,56 @@
+package de.dkfz.roddy.tools
+
+import groovy.transform.CompileStatic
+
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+@CompileStatic
+class DateTimeHelper {
+
+    private final DateTimeFormatter DATE_PATTERN
+
+    DateTimeHelper(String dateParserPatternPattern = null, ZoneId timeZoneId = ZoneId.systemDefault()) {
+        if (!dateParserPatternPattern)
+            DATE_PATTERN = DateTimeFormatter.ISO_DATE_TIME
+        else
+            this.DATE_PATTERN = DateTimeFormatter
+                    .ofPattern(dateParserPatternPattern)
+                    .withLocale(Locale.ENGLISH)
+                    .withZone(timeZoneId ?: ZoneId.systemDefault())
+    }
+
+    DateTimeFormatter getDatePattern() {
+        return DATE_PATTERN
+    }
+
+    ZonedDateTime parseTime(String str) {
+        ZonedDateTime date = ZonedDateTime.parse(str, DATE_PATTERN)
+        return date
+    }
+
+    static Duration toDuration(LocalDateTime ldt) {
+        // There is no plusYears, we just assume 365 days for a year
```

Comment:
Hm, very bad assumption. All output will be false. Also leap-seconds, etc. are ignored. This should not be called ```toDuration``` but ```toEstimatedDuration```, or a cleaner version should be implemented (which is admittedly complex and likely too complex for our purposes).

Reply:
Consider that Oracle advises to use Duration when time is measured in nanoseconds, while Period when time is measured in days, hours, etc.

https://docs.oracle.com/javase/tutorial/datetime/iso/period.html

Reply:
Thing is, I really did not figure out a better way. There are a lot of classes in java.time and you cannot use easily for what we need. It would be really nice to have something like DateTime1-DateTime2. But either I did not find it or it does not exist. That is, why I did it. I know the assumption is really bad and lacks a lot. But I also think, that for our case it should work. Our code is either using minutes (Roddy) or days (OTP), don't know how I can fix this in a good way.

Reply:
What about the Period class?

Reply:
The Period class does not allow us to use minutes or hours. It is based on days, weeks and years. I found out, that Duration has a between method which takes to LocalDateTime objects. This is a good start and I currently try to build things around it. Duration can also be negative which means you can just multiply it by -1 to get a positive duration.